libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -647,8 +648,14 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      */
     @Deprecated(message = "use a function withe the callback instead")
     fun resetTripSession() {
-        resetTripSession {
-            // no-op
+        runBlocking {
```

Comment:
I've read the comments why you've decided to do that and it seems reasonable but it still looks like we are introducing ANRs here. Could you leave a comment or sth that it's intentional?
Also, you have a typo in the deprecation message.

Reply:
Sure.

> the behavior is exactly the same as it used to be

I mean the comment for us, not for users. Because I'd find (and I found) `runBlocking` in production code quite suspicious.

Reply:
Am also not a fan of runBlocking here. If you run it from the main thread (@UiThread) then you will get your blocking result from Dispatchers.Main.immediate

Reply:
I had to push other PRs due to changelog conflicts anyway, so addressing this thread as well. Still open to alternatives for `runBlocking`.

Reply:
For reference and historical context pasting in here the original native impl as well:
```cpp
void NavigatorImpl::resetRideSession() {
  NNLOG_DEBUG_FN;

  std::promise<void> promise;
  reset([&promise] { promise.set_value(); });
  promise.get_future().wait();
}
```
That's why `runBlocking` seems appropriate to me.

Reply:
It's a deprecated function, so runBlocking is not that big of a deal.

I just find the new callback function overkill and confusing. If it is a simple function that needs to run on the main thread then it works and it's not as suspicious.

The new function, I'd just call it like this when calling from the main thread
```
mapboxNavigation.resetTripSession { } 
```

Reply:
I could understand the callback if we were using `Dispatchers.Main` instead of `Dispatchers.Main.immediate`

Because that changes the callback order here

This is `Dispatchers.Main`
```
// 1
mapboxNavigation.resetTripSession { 
 // 3
}
// 2
```

This is `Dispatchers.Main.immediate`
```
// 1
mapboxNavigation.resetTripSession { 
 // 2
}
// 3
```

Reply:
If the job Nav Native needs to do to reset the state is not straightforward and takes time or maybe even disk writes, etc. then a callback function definitely has it's place.

Otherwise, your simulated location might be ignored if you set it before the reset job is finished.

Reply:
@kmadsen `navigator.resetRideSession()` suspends.

So the flow that we have is this:
```
// 1
mapboxNavigation.resetTripSession { 
 // 2
 // suspend
 // 4
}
// 3
```

Reply:
oh right, resetRideSession is making it switch you're right. 

`runBlocking` is really our only choice here, that I know of too

Reply:
What do you mean exactly? The blocking function is already removed from Nav Native in the latest version, we just keep it in our public API for backwards compatibility.

Reply:
Updated https://github.com/mapbox/mapbox-navigation-android/pull/6675 to use the callback. resolving this thread